### ISSUE TITLE
Research: Specifications for a logo

### ISSUE THREAD

**Issue by mtlynch** [mtlynch/spaceduck#65]
I'd like to add a logo to fill the empty space on the lefthand side of the blog. I'd also like to replace the favicon image with a custom logo.

What are the recommendations in terms of image specs (size, transparency, background) for both images?

**Comment by shredtechular** [comment]
# Image specifications
The below write-up contains my recommendations in terms of image specifications for the left sidebar image in the parent theme as well as for favicons in general.

## Left sidebar background image
The parent theme's example sidebar background image is 1000px wide by 1500px tall.  As you can obviously predict, the sidebar is not ever going to be that wide or tall on most screens/devices and uses the `center / cover` background CSS value to ensure the image is resized to fill the entire container from the center out.  This results in a large portion of the sides being cut off to keep the original image ratio.  But it is also big enough where you won't ever lose continuity with the image on the sidebar.

It's a little difficult to give a simple answer to the question, because if you have something in mind and how you want that image to react responsively, then we can certainly alter the CSS to make sure that happens.  For example, if it is more important for the entire image to show no matter what, then we could change the CSS to `contain` instead of `cover` and that is when transparency and gradients could come into play so the edges of the background image blend well with the container's background.  I think it really just depends what you have in mind and how you want it to look in general though.

However, if you want to minimize, the CSS changes, then my recommendation for size is to make whatever you want to show in the sidebar the center of the image and have enough room on the edges where it won't matter if something is cut off a little.  1000px by 1500px is safe to make sure it will stretch on very large screens, but I think you could also easily get away with 800px by 1200px as well.  I would recommend keeping the 2:3 ratio if at all possible as well.  There is currently no transparency or gradient settings in the CSS itself, so if you are looking for any fade effects, then doing that in the image itself is totally fine, it may just add to the overall image size.  Otherwise, the text in the sidebar is obviously white so in order to provide a large enough contrast where the text is clearly readable, having a darker image is probably wise as well.  Unless you wanted to change the color of the sidebar text too.

## Favicon image
I have become a big fan of [https://realfavicongenerator.net/](https://realfavicongenerator.net/) for favicon images/code as it seems to do a lot of work for me when generating different sizes of these and making sure the code is correct for all devices.

As you can see on that site, it recommends a 260x260 image but 70x70 at a minimum in order to generate the other sized favicon images.  As far as transparency/background for favicons goes, I don't think there is any industry recommendation that I know of, I think it is personal preference and what you have in mind for the new favicon.  I personally like the simple favicons that are 1 or 2 colors total and do have a transparent background.  But many sites have a solid background color as well maybe to make sure their favicon is clearly visible no matter what color themes a user's browser may be using.

Hope this answers the questions sufficiently, please let me know if you need some other information.

**Comment by mtlynch** [comment]
Cool, I'll digest this and follow up with any questions.

**Comment by mtlynch** [comment]
One follow up question: what are our options for showing it on mobile? I suspect many readers may not ever click the menu button on mobile, but I'd like them to see the logo. Could we put it in a little header bar at the top? Not anything that stays visible after the user scrolls down, but just something small they see at the top, like here:

![image](https://user-images.githubusercontent.com/7783288/39090584-a60d52ec-45b0-11e8-97c6-bf790ab89a46.png)

**Comment by shredtechular** [comment]
That's doable, but will require a little customization to the theme, not a big deal though.  

Since the current mobile menu actually stays fixed at the top even as the user scrolls it would probably be easier to keep that current behavior if you are okay with the mobile "header bar" staying fixed at the top as the user scrolls down.  I can add some CSS to make it look better than it does now... 

## Current mobile header bar as user scrolls:
<kbd>
<img width="392" alt="screen shot 2018-04-21 at 10 33 33 pm" src="https://user-images.githubusercontent.com/2396774/39090742-4b0d1472-45b5-11e8-8e85-955234d5ff19.png">
</kbd>

## New header bar (something like this anyway...)
The idea here is to have a solid background for the mobile header bar so there isn't any "bleed" of hamburger menu and logo into the text...it just completely hides anything underneath.  So, you may not have as much real estate for reading on mobile, but I think this presents better anyway...
<kbd>
<img width="389" alt="screen shot 2018-04-21 at 10 33 07 pm" src="https://user-images.githubusercontent.com/2396774/39090707-63e70404-45b4-11e8-9293-111acb180c5f.png">
</kbd>

**Comment by mtlynch** [comment]
Nice, I like that. If I commissioned a separate version for that space, what specs do you recommend? It seems like the 2:3 ratio wouldn't work as well there.

**Comment by shredtechular** [comment]
Yeah, that's a good point...  Right now that space is about 90px height x ~85% screen width.  So, I'd say something like 75px x 250px would be a good guideline to work with.
